src/Components/Components/src/RenderTree/RendererInfo.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Provides information about the platform that the component is running on.
+/// </summary>
+/// <param name="rendererName">The name of the platform.</param>
+/// <param name="isInteractive">A flag to indicate if the platform is interactive.</param>
+public sealed class RendererInfo(string rendererName, bool isInteractive)
+{
+
+    /// <summary>
+    /// Gets the name of the platform.
+    /// </summary>
+    public string Name { get; } = rendererName;
```

Comment:
This is just academic, but I'm curious how this differs from:

```cs
public string Name => rendererName;
```

I would have thought doing it like this reduces the amount of storage required since we're reading the generated field directly instead of copying it into another field on construction, but maybe the compiler optimizes it to be the same regardless.

Not a big deal but if you happen to know then I'm interested!

Reply:
The compiler optimizes that afaik.

Reply:
Yeah I was curious and checked; the generated code is virtually identical. In both cases there's only one backing field.